daemon/algod/api/server/v2/generated/model/types.go

```diff
@@ -638,6 +638,9 @@ type SimulateRequest struct {
 	// AllowMoreLogging Lifts limits on log opcode usage during simulation.
 	AllowMoreLogging *bool `json:"allow-more-logging,omitempty"`
 
+	// ExecTraceConfig An object that configures simulation execution trace.
+	ExecTraceConfig *SimulateTraceConfig `json:"exec-trace-config,omitempty"`
+
```

Comment:
I'm commenting here, but I think it applies to all the fields of `SimulateRequest`, and I just hadn't thought about it before.
Since this is an input only structure, If we use values instead of pointers, won't the json deserialization give us what we want, without needing to worry about `nil` when we consume the data?  For example, if `AllowMoreLogging` was `bool` instead of `*bool` doesn't that make consuming code easier?
(Perhaps this structure _is_ used for output as well, as we report what the user asked for?)

Reply:
As @ahangsu pointed out in a different comment, this file is generated, so unfortunately we have no control here. The generator makes anything that's not required a pointer.

But you're right, using values would be sufficient (and simpler) for our use case.

Reply:
Jason made some good point, this is generated file, so we have minor control over...

It would indeed easier to use directly object rather than ptr, for we can at least get around a `!= nil` check.

Reply:
Actually, for the sake of simplicity and not copying twice, see [add logic for exec mode sensitive group index returning](https://github.com/algorand/go-algorand/pull/5360/commits/9640723afceda09747fdc6e40c9801027d8254eb), that I decided to use `codec` to avoid using this generated code. I want to get around the `nil` check for pointer, for fields are not strictly required.

Reply:
I believe so, that the `codec` thing should be the Algorand forked version for msgpack handling, with `omitempty` modified for slice, afaik.